src/electrum/electrumserver.cpp

```diff
@@ -0,0 +1,152 @@
+#include "electrum/electrumserver.h"
+#include "electrum/electrs.h"
+#include "util.h"
+
+#include <boost/process.hpp>
+#include <chrono>
+#include <string>
+
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+#include <signal.h>
+#endif
+
+using namespace boost::process;
+
+static bool send_kill_signal(pid_t id)
+{
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+    if (kill(id, SIGTERM) != 0)
+    {
+        LOGA("Electrum: Failed to send SIGTERM signal to server.");
+        return false;
+    }
+    return true;
+#else
+    LOGA("Electrum: Failed to send kill signal. Not implemented for this platform.");
+    return false;
+#endif
+}
+
+static void log_pipe(ipstream &pipe)
+{
+    std::string line;
+    while (pipe && std::getline(pipe, line))
+    {
+        LOG(ELECTRUM, "Electrum: %s", line);
+    }
+}
+
+//! give the program a second to complain about startup issues, such as invalid
+//! parameters.
+static bool startup_check(child &process)
+{
+    auto timeout = std::chrono::seconds(1);
+    if (!process.wait_for(timeout))
+    {
+        // process hasn't exited, good.
+        return true;
+    }
+    LOGA("Electrum: server exited with exit code: %d", process.exit_code());
+    return false;
+}
+
+static void log_args(const std::string &path, const std::vector<std::string> &args)
+{
+    if (!Logging::LogAcceptCategory(ELECTRUM))
+    {
+        return;
+    }
+
+    std::stringstream ss;
+    ss << path;
+    for (auto &a : args)
+    {
+        ss << a << " ";
```

Comment:
ss << " " << a;

Or you get and executable named "electrs-vvvv", and an extra space at the end I presume.

Reply:
Fixed